source/hic_hal/freescale/k20dx/gpio.c

```diff
@@ -98,7 +102,24 @@ void gpio_set_msc_led(gpio_led_state_t state)
 
 uint8_t gpio_get_sw_reset(void)
 {
+#if defined(RBL_nRF51)
+    uint8_t  status;
+    uint32_t pcr_value;
+    // save the configuration of GPIO
+    pcr_value = PIN_nRESET_PORT->PCR[PIN_nRESET_BIT];
+    // configure pin as GPIO input_pullup
+    PIN_nRESET_PORT->PCR[PIN_nRESET_BIT] = PORT_PCR_MUX(1) | PORT_PCR_PE_MASK | PORT_PCR_PS_MASK;
+    PIN_nRESET_GPIO->PDDR &= ~PIN_nRESET;
+    busy_wait(5);
+    status = (PIN_nRESET_GPIO->PDIR & PIN_nRESET) ? 1 : 0;
+    // restore the configuration
+    PIN_nRESET_GPIO->PDDR |= 1 << PIN_nRESET_BIT;
+    PIN_nRESET_PORT->PCR[PIN_nRESET_BIT] = pcr_value;
+    
+    return status;
+#else
```

Comment:
This code shouldn't be needed, since the RBL_nrf51 already has a pullup on the reset button according to this schematic:
http://redbearlab.com/nrf51822

Are you seeing problems without this code? If so can, you send me details on how to reproduce it?